### PR TITLE
keosd - Exit program if keosd fails to aquire wallet lock

### LINKED ISSUE #513: if keosd fails to aquire wallet lock it lingers forever
If keosd fails to acquire the wallet lock it will print a failure message but just linger forever until ctrl-C. It should just exit in this case.
```
$ programs/keosd/keosd
info  2022-11-24T02:04:41.639 keosd     wallet_plugin.cpp:33          plugin_initialize    ] initializing wallet plugin
warn  2022-11-24T02:04:41.639 keosd     wallet_plugin.cpp:51          plugin_initialize    ] 3120000 wallet_exception: Wallet exception
Failed to lock access to wallet directory; is another keosd running?
    {}
    keosd  wallet_manager.cpp:304 initialize_lock

Failed to initialize
```
Seen on main, unknown if it affects 3.x

### PR DESCRIPTION
**Pull request by greg7mdp**
Resolves #513 

Should we backport to 3.2?

### CODE REVIEW

File: programs/keosd/main.cpp [hunk 1]
```diff
@@ -93,6 +93,7 @@ int main(int argc, char** argv)
              opts.count("print-default-config")) {
             return 0;
          }
+         return EXIT_FAILURE;
```
Comment (heifner) [inline programs/keosd/main.cpp:96]: Seems like this should match `nodeos` `main.cpp`, which would be to return `-1` (`INITIALIZE_FAIL`).

**Review by heifner — CHANGES_REQUESTED** [review]
Should be backported to 3.1 & 3.2.

File: programs/keosd/main.cpp (hunk 1 above)
Comment (greg7mdp) [reply programs/keosd/main.cpp:96]: Sure, will do, although `EXIT_FAILURE` is a [platform standard]( https://en.cppreference.com/w/c/program/EXIT_status) while `INITIALIZE_FAIL` is a leap invention :-)

Comment (greg7mdp) [reply programs/keosd/main.cpp:96]: @heifner do you think this should go to 3.2 first?

Comment (heifner) [reply programs/keosd/main.cpp:96]: I assume it is applicable to 3.1, so it goes there first then merged up to 3.2 and then to main.

Comment (greg7mdp) [reply programs/keosd/main.cpp:96]: OK, should I duplicate the whole enum in keosd/main.cpp?

```
enum return_codes {
   OTHER_FAIL        = -2,
   INITIALIZE_FAIL   = -1,
   SUCCESS           = 0,
   BAD_ALLOC         = 1,
   DATABASE_DIRTY    = 2,
   FIXED_REVERSIBLE  = SUCCESS,
   EXTRACTED_GENESIS = SUCCESS,
   NODE_MANAGEMENT_SUCCESS = 5
};
```

It it was me I'd have it in a common header, but I don't know which one.

Comment (heifner) [reply programs/keosd/main.cpp:96]: Maybe `plugins/chain_interface/include/eosio/chain/plugin_interface.hpp` ?

**Pull request closed without merging**